### DISCUSSION TITLE
Bruno GitHub Actions

### DISCUSSION THREAD

**Discussion by krummbar** [usebruno/bruno#2635] [Ideas]
Hi,

I tried reaching out to @helloanoop via Discord about this, but haven't received a response yet. I thought maybe discussions would be a better place to address this.

I've created two GitHub Actions, that may be useful in CI pipelines.

https://github.com/krummbar/bruno-run-action
A containerized action, executing the bru run cli. Basically a simple Docker wrapper that maps GitHub Action inputs to the bruno's command line interface.

https://github.com/krummbar/bruno-run-report
This action is able to parse the JSON results of a bru run command and attaches a markdown summary to the action itself. So it is closely linked to the previous action.

I have two questions regarding those actions.

1. Before publishing this to the GitHub marketplace, I wanted to ask if this is something you want to have officially under the bruno domain, or would it be fine if I simply publish it there?
2. The **bruno-run-report** internally uses [`jq`](https://jqlang.github.io/jq/) to transform the bru JSON output to a markdown summary. I've started simple here, but it quickly reached complexity that makes it hard to maintain. I was thinking about reimplementing this as JS/TS action. Before doing that I wanted to ask if it would be feasible to provide markdown as output format out-of-the-box within the runner? E.g. `bru run --format markdown --output bru-run-summary.md`

**Comment by abhishek-bruno** [comment]
Update for everyone following this: the official Bruno GitHub Action has now shipped.

Marketplace: https://github.com/marketplace/actions/bruno-cli
Repo: https://github.com/usebruno/bruno-cli-action
Docs: https://docs.usebruno.com/bru-cli/github-actions/overview

@krummbar, to answer your original questions: yes, we wanted this officially under the Bruno org, and your work was the precursor that kicked it off. Thank you for that. The action installs @usebruno/cli, runs your bru command and exposes machine-readable outputs (passed, failed, total, duration-ms, exit-code) for downstream steps. Rendering (PR comments, annotations, step summary) is delegated to ecosystem actions like EnricoMi and dorny, so the markdown-summary need is covered without baking it into the runner.

We'd love for you to try it and contribute. If you spot gaps or have ideas, please open an issue or PR on the repo.
